src/industry_cmd.cpp

```diff
@@ -2425,12 +2426,21 @@ static void UpdateIndustryStatistics(Industry *i)
 				i->last_prod_year = TimerGameCalendar::year;
 				pct = std::min(i->this_month_transported[j] * 256 / i->this_month_production[j], 255);
 			}
-			i->last_month_pct_transported[j] = pct;
+			for (size_t k = lengthof(i->past_pct_transported) - 1; k >= 1; k--){
+				i->past_pct_transported[k][j] = i->past_pct_transported[k-1][j];
+			}
+			i->past_pct_transported[0][j] = pct;
 
-			i->last_month_production[j] = i->this_month_production[j];
+			for (size_t k = lengthof(i->past_production) - 1; k >= 1; k--){
+				i->past_production[k][j] = i->past_production[k-1][j];
+			}
+			i->past_production[0][j] = i->this_month_production[j];
 			i->this_month_production[j] = 0;
 
-			i->last_month_transported[j] = i->this_month_transported[j];
+			for (size_t k = lengthof(i->past_transported) - 1; k >= 1; k--){
+				i->past_transported[k][j] = i->past_transported[k-1][j];
+			}
+			i->past_transported[0][j] = i->this_month_transported[j];
```

Comment:
Missing spaces in `[k - 1]` each time.

Switching the array around from `[month][slot]` to `[slot][month]` would give simpler code here I think -- although on second thoughts that would make the saveload changes a nightmare!